src/logging.rs

```diff
@@ -213,6 +215,41 @@ fn send_memfd_payload(sock: UnixDatagram, data: &[u8]) -> Result<usize, SdError>
     Ok(data.len())
 }
 
+fn fd_has_device_and_inode(fd: RawFd, device: dev_t, inode: ino_t) -> bool {
```

Comment:
What about returning `Result<bool>`, so that the caller is able to do some proper logging before mapping to `false`?